tty/usbacm/usbacm.c

```diff
@@ -297,9 +430,18 @@ static int usbacm_handleDeletion(usb_deletion_t *del)
 		next = dev->next;
 		if (dev->instance.bus == del->bus && dev->instance.dev == del->dev &&
 				dev->instance.interface == del->interface) {
+			fprintf(stderr, "usbacm: Device removed: %s\n", dev->path);
 			remove(dev->path);
 			LIST_REMOVE(&usbacm_common.devices, dev);
-			fprintf(stderr, "usbacm: Device removed: %s\n", dev->path);
+			if (dev->fifo != NULL) {
+				/* Non-blocking device */
+				while (dev->rxRunning)
```

Comment:
This is not good solution, there is a race condition - rxRunning can be already 0, but thread has not yet ended. You should do thread join here instead of this flag, this will also take care of waiting for thread to finish it's business.